File: greedy/python/b12931.py
from sys import stdin
N = int(stdin.readline())
nums = list(map(int, stdin.readline().split()))

flag = True
cnt = 0
while flag:
    flag &= False
    for i in range(N):
        if nums[i] > 1:
            cnt += nums[i] % 2
            nums[i] = nums[i] >> 1
            flag |= True
    if flag:
        cnt += 1

for i in range(N):
    if nums[i] > 0:
        cnt += 1
print(cnt)

File: greedy/cpp/b12931.cpp
#include <iostream>

int N;
int nums[50];

bool check() {
	for (int i = 0; i < N; ++i) {
		if (nums[i] > 1) {
			return true;
		}
	}
	return false;
}

int main() {
	std::cin >> N;
	for (int i = 0; i < N; ++i) {
		std::cin >> nums[i];
	}

	int cnt = 0;
	while (check()) {
		for (int i = 0; i < N; ++i) {
			if (nums[i] > 1) {
				cnt += nums[i] % 2;
				nums[i] = nums[i] / 2;
			}
		}
		cnt += 1;
	}

	for (int i = 0; i < N; ++i) {
		if (nums[i]) {
			cnt += 1;
		}
	}
	std::cout << cnt << std::endl;
}